backend/services/frontend/handler/frontend.go

```diff
@@ -969,3 +973,57 @@ func (e *Frontend) SrsUserDueHandler(c *fiber.Ctx) error {
 		DueDecks: dueCards.DueDecks,
 	})
 }
+
+func (e *Frontend) SubscribeNotificationsHandler(c *fiber.Ctx) error {
+	userID := helper.GetUserIDFromContext(c)
+	subscription := &pbCommon.PushSubscription{}
+	if err := c.BodyParser(subscription); err != nil {
+		return err
+	}
+	if err := helper.EnsureNotEmpty(subscription.Endpoint, subscription.Auth, subscription.P256Dh); err != nil {
+		return err
+	}
+
+	rspSubscribeNotifications, err := e.notificationsService.Subscribe(c.Context(), &pbCommon.PushSubscriptionRequest{
+		UserID: userID,
+		Subscription: &pbCommon.PushSubscription{
+			Endpoint: subscription.Endpoint,
+			Auth:     subscription.Auth,
+			P256Dh:   subscription.P256Dh,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return c.SendString(rspSubscribeNotifications.SubscriptionID)
+}
+
+func (e *Frontend) GetUserNotificationSubscriptionsHandler(c *fiber.Ctx) error {
+	userID := helper.GetUserIDFromContext(c)
+	rspGetUserSubscriptions, err := e.notificationsService.GetUserNotificationSubscriptions(c.Context(), &pbCommon.User{UserID: userID})
+	if err != nil {
+		return err
+	}
+	ids := make([]string, len(rspGetUserSubscriptions.Subscriptions))
+	for i, subscription := range rspGetUserSubscriptions.Subscriptions {
+		ids[i] = subscription.SubscriptionID
+	}
```

Comment:
you can create a `map` function for stuff like this (I think there might be already one? I think I saw this multiple times)

```go
// helper/map.go
func Map[I, O any](input []I, mapper func(I) O) []O {
	out := make([]O, len(input))
	for i, v := range input {
		out[i] = mapper(v)
	}
	return out
}

// handler/frontend.go
// ...
ids := helper.Map(rspGetUserSubscriptions.Subscriptions, func(s model.PushSubscription) {
	return s.SubscriptionID
})
```

Reply:
@TomRomeo you marked this as resolved because you don't want to implement it, or was this an accident?

Reply:
```suggestion
	ids := converter.ConvertToTypeArray(rspGetUserSubscriptions.Subscriptions, converter.ProtoNotificationSubscriptionToIDStringConverter)
	return c.JSON(converter.FiberGetUserSubscriptionsResponseBody{UserSubscriptions: ids})
```
There is the `ConvertToTypeArray` function in the converter package in place. And also, I would not just return an array in the response. It requires the following additions in the `typeConverter` and `fiberSerializerTypes` (not tested):
```go
func ProtoNotificationSubscriptionToIDStringConverter(subscription *pbCommon.PushSubscription) string {
	return subscription.SubscriptionID
}
```
```go
type FiberGetUserSubscriptionsResponseBody struct {
	UserSubscriptions []string `json:"userSubscriptions"`
}
```